ui-package/src/components/SubmitDepositsForm/DepositEntry.tsx

```diff
@@ -79,6 +87,109 @@ const DepositEntry = (props: IDepositEntryProps): React.ReactElement => {
             </Modal.Footer>
           </Modal>
         )}
+        {showTxDetails && (
+          <Modal show={true} onHide={() => setShowTxDetails(false)} size="lg" className="deposit-txs-modal">
+            <Modal.Header closeButton>
+              <Modal.Title>Initiated Deposits</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+              <p>Some deposits have already been submitted for this validator:</p>
+              {props.deposit.depositTxs.map((tx, index) => (
+                <div key={index + "-" + tx.tx_hash} className="mt-2">
+                  {index > 0 && <hr />}
+                  <div className="d-flex">
+                    <div className="tx-details-label">
+                      Tx Hash:
+                    </div>
+                    <div className="tx-details-value">
+                      <div className="d-flex">
+                        {props.explorerUrl ?
+                          <a className="flex-grow-1 text-truncate" href={props.explorerUrl + "tx/" + tx.tx_hash} target="_blank" rel="noreferrer">{tx.tx_hash}</a>
+                        : <span className="flex-grow-1 text-truncate">{tx.tx_hash}</span>
+                        }
+                        <div className="ms-2">
+                          <i className="fa fa-copy text-muted p-1" role="button" data-bs-toggle="tooltip" data-clipboard-text={tx.tx_hash} title="Copy to clipboard"></i>
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                  <div className="d-flex">
+                    <div className="tx-details-label">
+                      Block:
+                    </div>
+                    <div className="tx-details-value">
+                      <div className="d-flex">
+                        <span className="flex-grow-1 text-truncate">{tx.block}</span>
```

Comment:
Maybe a link to the block explorer and respective block number would be a nice to have.

Reply:
Nevermind. I just saw you have a link to the TX if the explorerUrl exists.